Publication Landscape and Document parser update

danich1 opened the pull request:
This PR is generated in response to the conversation in #81. The document parser was not converting words into their base form (lemmatization e.g. walked -> walk) and it was suggested that we incorporate that for generating document vectors. As a result the publication landscape for the server changed along with the SAUCIE model and publication paper dataset (uploaded to google cloud in a folder called updated-data). The KNN model will need to be re-sharded again as a heads up @dongbohu.

vincerubinetti commented:
It looks like the bin width has changed here.

![image](https://user-images.githubusercontent.com/8326331/114906705-50e39e00-9de8-11eb-8752-929646caa055.png)

I forget how we did this, but there's some variable in the javascript code you have to match to whatever the bin width was when you exported the plot.

vincerubinetti commented:
Here it is: https://github.com/greenelab/preprint-similarity-search/blob/master/frontend/src/map.js#L13

vincerubinetti commented:
Bin size of `0.65` doesn't look right. It looks like it's more like `0.4`:
![image](https://user-images.githubusercontent.com/8326331/114915749-f7806c80-9df1-11eb-9e30-9aa65115004c.png)

Also you ran the plot through `filter-plot.html` right? I think you did otherwise it wouldn't show anything, but on Netlify the graph was taking a while to load and the filter was supposed to fix that.

Is this plot data going to be updated more frequently now? If so I will probably want to figure out a way to pre-process this automatically.

danich1 commented:
>Bin size of 0.65 doesn't look right. It looks like it's more like 0.4:

Ah. Right now I understand why we were using bin_width. I changed my code to use R's auto binwidth generator, but that's not right. Will default back for that plot. It will be 0.85 binwidth.

>Also you ran the plot through filter-plot.html right? I think you did otherwise it wouldn't show anything, but on Netlify the graph was taking a while to load and the filter was supposed to fix that.

Yeah I ran the new plot through your filter-plot html code and updated the plot.json as well as the main json plot file as well.

dongbohu commented on server/word_vectors.py:
```diff
@@ -49,17 +49,22 @@ def parse_content(content, xml_file=True):
+                lambda x: str(x.lemma_),
+                filter(
+                    lambda tok: str(tok.lemma_) in model.wv
+                    and str(tok.lemma_) not in nlp.Defaults.stop_words,
```
Isn't `tok.lemma_` already a string?

danich1 replied:
Yes good call.

dongbohu commented on server/word_vectors.py:
```diff
@@ -70,12 +75,11 @@ def parse_content(content, xml_file=True):
-                word_model_wv[tok]
-                for tok in tokens
-                if tok in word_model_wv and tok not in nlp.Defaults.stop_words
+                word_model_wv[str(tok.lemma_)]
```
`str()` conversion seems redundant.

dongbohu commented on server/word_vectors.py:
```diff
@@ -70,12 +75,11 @@ def parse_content(content, xml_file=True):
-                if tok in word_model_wv and tok not in nlp.Defaults.stop_words
+                word_model_wv[str(tok.lemma_)]
+                for tok in nlp(page.getText())
+                if str(tok.lemma_) in word_model_wv and str(tok.lemma_) not in nlp.Defaults.stop_words
```
Redundant `str()` conversion.

dongbohu reviewed (commented):
Left a few comments.

dongbohu commented:
I am merging/closing this PR so that the frontend can use the latest plot json file.
